src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -3,13 +3,21 @@
 # Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
 #
 from enum import Enum
-from typing import Any, Dict, Iterable, List, Optional
+from typing import Any, Dict, List, Optional
 
 import snowflake.snowpark
 from snowflake.snowpark._internal.analyzer.expression import Attribute, Expression
 from snowflake.snowpark.row import Row
 from snowflake.snowpark.types import StructType
 
+# Python 3.8 needs to use typing.Iterable because collections.abc.Iterable is not subscriptable
+# Python 3.9 can use both
+# Python 3.10 needs to use collections.abc.Iterable because typing.Iterable is removed
+try:
+    from typing import Iterable
+except ImportError:
+    from collections.abc import Iterable
```

Comment:
I'm a fan of doing these a different way. See [pyupgrade's docs](https://github.com/asottile/pyupgrade#python2-and-old-python3x-blocks). Doing it this way we can rely on `pyupgrade` to automatically remove these blocks and we know exactly where the code goes for different Python versions.